File: banda/AMarotoChivite-banda/AMarotoChivite-banda-Java/src/main/java/simulacionBandaMusica/factories/CantanteFactory.java
package simulacionBandaMusica.factories;

import simulacionBandaMusica.models.Cantante;

import static simulacionBandaMusica.utils.sorteoDatos.*;

public class CantanteFactory {

    public static Cantante create() {
        String name = randomName();
        int experienceYear = randomAnno();
        int pulmonarCapacity = randomPulmonarCapacity();
        return new Cantante(name, "Voz", experienceYear, pulmonarCapacity);
    }

}

File: banda/AMarotoChivite-banda/AMarotoChivite-banda-Kotlin/src/main/kotlin/simulacionBandaMusica/factories/CantanteFactory.kt
package simulacionBandaMusica.factories

import simulacionBandaMusica.models.Musico
import simulacionBandaMusica.utils.sorteoDatos

object CantanteFactory {
    fun create(): Musico.Cantante {
        val name = sorteoDatos.randomName()
        val experienceYear = sorteoDatos.randomAnno()
        val pulmonarCapacity = sorteoDatos.randomPulmonarCapacity()
        return Musico.Cantante(name, "Voz", experienceYear, pulmonarCapacity)
    }
}